src/reducers/user.js

```diff
@@ -1,9 +1,15 @@
 import types from '../actions';
 
-const user = (state = {}, action) => {
+const defaultUser = {
+  attributes: { moderator: false },
+};
+
+const user = (state = defaultUser, action) => {
   switch (action.type) {
     case types.user.set.name:
       return { ...state, username: action.username };
+    case types.user.set.as.moderator:
+      return { username: action.username, attributes: { ...state.attributes, moderator: true } };
```

Comment:
should you do?

```javascript
...state, username: action.username, attributes: { moderator: true } }
```

This way if more default values are added on L3, then this will be applied to the moderator as well, and I think this would follow the pattern on L10

Reply:
If we use `attributes: { moderator: true }` we are ignoring all the future attributes that the user will have because `attributes: { moderator: true }` is creating a new object with the moderator key only and that is not the intention here.

Reply:
correct, but the same thing will happen for all the properties that you will add to the base user, what if you do?: 

```javascript
{...state, username: action.username, attributes: { ...state.attributes, moderator: true } }

Reply:
You are right!